### ISSUE TITLE
Localsend shows raw DioException when receiving device goes offline

### ISSUE THREAD

**Issue by tusharlock10** [localsend/localsend#1152]
### Describe the bug

Getting an error when the receiving device goes offline (either from closes Localsend or goes off the network). It shows the raw DioException message, instead of something more pleasant/ useful eg. '<device_name> has gone offline'.

Error - DioException [connection error]: The connection errored: Connection refused This indicates an error which most likely cannot be solved by the library.
Error: SocketException: Connection refused (OS Error: Connection refused, errno = 61), address = 192.168.1.4, port = 51771

### To reproduce

1. Open Localsend on Device1 (sender) and Device2 (receiver) on same network
2. Device2 should now be showing on Device1 to send files
3. Select something to send on Device1
4. Disconnect or close Localsend on Device2
5. Send files on Device2 from Device1
6. An error will appear when trying to send files, which is the same error shown above

### Expected behavior

Two solutions could be employed - 
1. Show a useful message eg. 'Device2 has disconnected'
2. Remove Device2 from the list of available devices to send, as soon as it its disconnected 

### Screenshots

<img width="931" alt="Screenshot 2024-02-22 at 7 04 41 PM" src="https://github.com/localsend/localsend/assets/36128584/2889bb78-c201-40af-a940-787296e58baf">

### Desktop (please complete the following information)

MacBook Pro M2 16GB

### Smartphone (please complete the following information)

Nothing Phone One

**Comment by SonicWarrior1** [comment]
Yes i have been able to reproduce this issue . It is occuring because the application requires manual refreshing to update the list of available devices, leading to confusion when devices go offline without being removed from the list.

Solution Proposal:

1. Implement Periodic Scanning: Enable the app to periodically scan for nearby devices, ensuring the list stays current without manual intervention.
2. Check Device Connectivity: Verify the online/offline status of devices before sending, preventing users from attempting to connect to offline devices.

Also such fromat of error messages are irrelevant to the users so it should be displayed as some simpler messages.

**Comment by SFNsansub** [comment]
I was facing the same issue and the solution purposed by @SonicWarrior1 wasnt working. 
The solution was to reboot my router for new local ip addresses which seems to work out.
